saml-core/src/main/java/org/keycloak/saml/processing/core/parsers/saml/assertion/SAMLAttributeValueParser.java

```diff
@@ -105,40 +108,50 @@ public Object parse(XMLEventReader xmlEventReader) throws ParsingException {
             return StaxParserUtil.getElementText(xmlEventReader);
         }
 
-        // KEYCLOAK-18417: Simply ignore unknown types
-        logger.debug("Skipping attribute value of unsupported type " + typeValue);
-        StaxParserUtil.bypassElementBlock(xmlEventReader);
-        return null;
+        if (typeValue.contains(":")) {
+            return parseAsString(xmlEventReader, element);
+        } else {
+            // KEYCLOAK-18417: Simply ignore unknown types
+            logger.debug("Skipping attribute value of unsupported type " + typeValue);
+            return null;
```

Comment:
While the old code called `StaxParserUtil.bypassElementBlock(xmlEventReader);` whenever there was something unknown, the new code doesn't call it any more. I suspect this could lead to a wrong result. Please have a look.

Reply:
As we can now parse anything (even without a prefix separator) I removed the else-branch